3 - Assignment/240326 - Rust Basic + Roguelike Game, Week 1/collatz-sequence/src/lib.rs

```diff
@@ -1,6 +1,16 @@
 /// Determine the length of the collatz sequence beginning at `n`.
 pub fn collatz_length(mut n: i32) -> u32 {
-    todo!("Implement this")
+    let mut count = 1;
+    while n > 1{
+        count += 1; 
+        if n % 2 == 0 {
+            n /= 2;
+        }
+        else{
+            n = 3*n+1;
+        }
+    }
```

Comment:
`cargo fmt`를 통해 코드 스타일을 맞춰주세요.

```suggestion
    while n > 1 {
        count += 1; 
        if n % 2 == 0 {
            n /= 2;
        } else {
            n = 3 * n + 1;
        }
    }
```

Reply:
코드를 다음처럼 간단하게 작성할 수도 있습니다.
```suggestion
    while n > 1 {
        n = if n % 2 == 0 { n / 2 } else { 3 * n + 1 };
        count += 1;
    }
```